bouncer/src/repo/core/model/bson/repo_bson_factory.h

```diff
@@ -288,15 +265,22 @@ namespace repo {
 				static MeshNode makeMeshNode(
 					const std::vector<repo::lib::RepoVector3D>        &vertices,
 					const std::vector<repo_face_t>                    &faces,
-					const std::vector<repo::lib::RepoVector3D>        &normals = std::vector<repo::lib::RepoVector3D>(),
-					const std::vector<std::vector<float>>             &boundingBox = std::vector<std::vector<float>>(),
-					const std::vector<std::vector<repo::lib::RepoVector2D>>   &uvChannels = std::vector<std::vector<repo::lib::RepoVector2D>>(),
-					const std::vector<repo_color4d_t>                 &colors = std::vector<repo_color4d_t>(),
-					const std::vector<float>                          &ids = std::vector<float>(),
-					const std::string                                 &name = std::string(),
-					const std::vector<repo::lib::RepoUUID>            &parents = std::vector<repo::lib::RepoUUID>(),
+					const std::vector<repo::lib::RepoVector3D>        &normals,
+					const std::vector<std::vector<float>>             &boundingBox,
+					const std::vector<std::vector<repo::lib::RepoVector2D>>   &uvChannels,
+					const std::vector<repo_color4d_t>                 &colors,
+					const std::string                                 &name,
+					const std::vector<repo::lib::RepoUUID>            &parents,
 					const int                                         &apiLevel = REPO_NODE_API_LEVEL_1);
 
+				static MeshNode makeMeshNode(
+					const std::vector<repo::lib::RepoVector3D>& vertices,
+					const std::vector<repo_face_t>& faces,
+					const std::vector<repo::lib::RepoVector3D>& normals,
+					const std::vector<std::vector<float>>& boundingBox,
+					const std::vector<std::vector<repo::lib::RepoVector2D>>& uvChannels,
+					const std::vector<repo_color4d_t>& colors);
```

Comment:
looking at this it makes me wonder if we should still pretend that we support colors on a mesh when it doesn't get used anywhere.... 😆

Reply:
we're also adding `{}` unncessarily everywhere to accomodate it...

I think we should either just remove it from the function, or supply a default value?

Reply:
Colour support is finally removed 😆